### ISSUE TITLE
Reordering of targeting rules does not work

### ISSUE THREAD

**Issue by kamilbiela** [markphelps/flipt#835]
**Describe the bug**

First of all thank you for this software, looks to be exactly what I need. 
Can't save new order for targeting rules.  UI works ok, but changes are not persisted.

**Version Info**

1.8.1 - fetched as `latest` docker image tag

**To Reproduce**

add rules to targeting, try to reorder them, click "Reorder Rules" button.
Message "Invalid field segmentkey: must not be empty" shows up.

docker log: 
```
INFO[3440] finished unary call with code InvalidArgument  error="rpc error: code = InvalidArgument desc = invalid field segmentKey: must not be empty" grpc.code=InvalidArgument grpc.method=UpdateRule grpc.service=flipt.Flipt grpc.start_time="2022-04-26T10:54:16Z" grpc.time_ms=0.039 peer.address="127.0.0.1:33384" server=grpc span.kind=server store=sqlite system=grpc
```

After refresh targeting rules are not reordered

**Expected behavior**

Ordering is persisted 

**Screenshots**

![image](https://user-images.githubusercontent.com/442972/165285244-85716583-66c1-44cd-8917-db85fb23976a.png)

After clicking reorder button:
![image](https://user-images.githubusercontent.com/442972/165286114-0fcf50d8-c94b-419e-bbe0-53b90e207aa7.png)
![image](https://user-images.githubusercontent.com/442972/165286145-3c0fbbc3-96d7-4f58-8e84-c900a4db9d9d.png)

**Attach an export**

I don't think I can do it, running ad hoc docker container to test functionalities with --rm flag.

**Additional context**

I've added rules, deleted, added more and reordered them. 

Examples:

running it from docker latest tag with command from documentation just to evaluate it.

**Comment by markphelps** [comment]
Thanks for the bug report @kamilbiela !

I think I have a fix in #836. Will need to add a new integration test first, but assuming that succeeds I should be able to release a new version `1.8.2` with the fix tonight

**Comment by kamilbiela** [comment]
Checked on 1.8.2 - works perfectly, thank you!
